Source/BasicMathFunctions/arm_maximum_s8.c

```diff
@@ -0,0 +1,277 @@
+/*
+ * SPDX-FileCopyrightText: Copyright 2024 Arm Limited and/or its affiliates <open-source-office@arm.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* ----------------------------------------------------------------------
+ * Project:      CMSIS NN Library
+ * Title:        arm_maximum_s8
+ * Description:  Minimum and Maximum
+ *
+ * $Date:        08 October 2024
+ * $Revision:    V.1.0.0
+ *
+ * Target :  Arm(R) M-Profile Architecture
+ *
+ * -------------------------------------------------------------------- */
+
+#include "arm_nnfunctions.h"
+#include "arm_nnsupportfunctions.h"
+
+/**
+ *  @ingroup Public
+ */
+
+/**
+ * @addtogroup minimumMaximum
+ * @{
+ */
+
+static arm_cmsis_nn_status
+arm_max_no_broadcast_s8(const int8_t *input_1, const int8_t *input_2, int8_t *output, int32_t flat_size)
+{
+#if defined(ARM_MATH_MVEI)
+    while (flat_size >= 16)
```

Comment:
Why split this into two sections? Why not have one section and use predicates?

Reply:
A bit of a habit, but from past testing with other operators usually calculating the predication mask and using the predicated instructions in every iteration lowers performance and often gets a bit awkward with pointer incrementing, but I guess since these pointers are just copies and the incrementing is linear then that part of the argument is not really valid in this case

Reply:
Did some testing and in this case performance was the same so reduced down to a only a predicated instruction loop